### PR TITLE
Add Teleport name changes

### PR DESCRIPTION
**Pull request by Malfuryent**
From the feature request : "Changing name of Sceptre teleports #3354"
I have created this plugin which changes unclear teleport locations to a more understandable format.

https://gyazo.com/6ce3ca7c3996f896c4a0d90dbd1ef59d

These are the teleports that have currently been changed. 
If there are more which could use a different alias feel free to let me know!

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesConfig.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+import net.runelite.client.config.Config;
+import net.runelite.client.config.ConfigGroup;
+
+
+@ConfigGroup(
+	keyName = "teleportnamechanges",
+	name = "Teleport Name Changes",
+	description = "Changes teleport names to be more understandable"
+)
+public interface TeleportNameChangesConfig extends Config
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesConfig.java]: You don't need a config if it doesn't have any options in it

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+import net.runelite.api.*;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.client.config.ConfigManager;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import javax.inject.Inject;
+
+@Slf4j
+@PluginDescriptor(
+	name = "Rename unclear teleport locations "
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: Space at the end. Also this should be a shorter name.

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 3]
```diff
@@ -0,0 +1,125 @@
...
+)
+public class TeleportNameChangesPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Provides
+	TeleportNameChangesConfig provideConfig(ConfigManager configManager)
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: This isn't required for a plugin

**Comment by MESLewis** [comment]
You should squash your git history to be clean

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesConfig.java (hunk 1 above)
Comment (Malfuryent) [reply runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesConfig.java]: So i can delete the whole config and just leave the plugin?

Comment (SoyChai) [reply runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesConfig.java]: Yup, a Plugin doesn't need a Config to run.

**Comment by Malfuryent** [comment]
Should be allright now 👍

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+ * LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ * ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.teleportnamechanges;
+
+import com.google.common.eventbus.Subscribe;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.*;
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: Don't use * imports

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+@PluginDescriptor(
+	name = "Rename teleports"
+)
+public class TeleportNameChangesPlugin extends Plugin
+{
+	@Inject
+	private Client client;
+
+	@Subscribe
+	// Pharaoh staff renaming
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: This comment looks like it should be inside the function

**Comment by MESLewis** [comment]
Squash all the commits into one

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+			{
+				if (object.getTarget().equals("<col=ff9040>Kharyrll teleport"))
+				{
+					object.setTarget("<col=ff9040>Canifis teleport");
+				}
+
+				if (object.getTarget().equals("<col=ff9040>Senntisten teleport"))
+				{
+					object.setTarget("<col=ff9040>Digsite teleport");
+				}
```
Comment (MESLewis) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: New line after each block

**Comment by MESLewis** [comment]
A commit message should be added and the title made more meaningful of the purpose of the PR

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 7]
```diff
@@ -0,0 +1,121 @@
...
+import net.runelite.api.MenuEntry;
+import net.runelite.api.Client;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import javax.inject.Inject;
+
+@Slf4j
+@PluginDescriptor(
+	name = "Rename teleports"
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: Maybe "Menu Entry Renamer" to follow the Menu Entry Swapper plugin?

File: runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+package net.runelite.client.plugins.teleportnamechanges;
+
+import com.google.common.eventbus.Subscribe;
+import lombok.extern.slf4j.Slf4j;
+import net.runelite.api.MenuEntry;
+import net.runelite.api.Client;
+import net.runelite.api.events.MenuEntryAdded;
+import net.runelite.client.plugins.Plugin;
+import net.runelite.client.plugins.PluginDescriptor;
+import javax.inject.Inject;
```
Comment (raqes) [inline runelite-client/src/main/java/net/runelite/client/plugins/teleportnamechanges/TeleportNameChangesPlugin.java]: Should probably move this javax import up to match the import order of other plugins.

**Comment by raiyni** [comment]
This is covered by the Easy Teleports hub plugin now.

**Pull request closed without merging**